### PR TITLE
PYIC-7048: read bearer token ttl from config

### PR DESCRIPTION
**Pull request by sam803**
### What changed

Read bearer token ttl from the config rather than an env var

### Why did it change

To be consistent with other related config variables

### Issue tracking

- https://govukverify.atlassian.net/browse/PYIC-7048

### Environment variables or secrets

- [X] No environment variables or secrets were added or changed

### Other considerations

- [X] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java [hunk 1]
```diff
@@ -114,11 +112,14 @@ private String formatPath(String path, String... pathProperties) {
...
-                .map(Long::valueOf)
-                .orElse(DEFAULT_BEARER_TOKEN_TTL_IN_SECS);
+        try {
+            return Long.parseLong(getParameter(ConfigurationVariable.BEARER_TOKEN_TTL));
+        } catch (ConfigParameterNotFoundException | NumberFormatException e) {
+            LOGGER.warn(
+                    LogHelper.buildErrorMessage("Failed to get bearer token ttl from config", e));
+            return DEFAULT_BEARER_TOKEN_TTL_IN_SECS;
+        }
     }
```
Comment (Joe-Edwards-GDS) [inline libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java]: I think I'd prefer removing this method entirely, and either doing `Long.parseLong(getParameter(...))` where it's used, or adding a more generic `getLongParameter(..)` which does the parsing for you.

As a general rule, I think we should avoid adding specific methods for each bit of config (or we should add specific methods for _all_ config values, but that's very tedious).

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java [hunk 2]
```diff
@@ -114,11 +112,14 @@ private String formatPath(String path, String... pathProperties) {
...
     public long getBearerAccessTokenTtl() {
-        return Optional.ofNullable(getEnvironmentVariable(BEARER_TOKEN_TTL))
-                .map(Long::valueOf)
-                .orElse(DEFAULT_BEARER_TOKEN_TTL_IN_SECS);
+        try {
+            return Long.parseLong(getParameter(ConfigurationVariable.BEARER_TOKEN_TTL));
+        } catch (ConfigParameterNotFoundException | NumberFormatException e) {
+            LOGGER.warn(
+                    LogHelper.buildErrorMessage("Failed to get bearer token ttl from config", e));
+            return DEFAULT_BEARER_TOKEN_TTL_IN_SECS;
```
Comment (Joe-Edwards-GDS) [inline libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java]: Do we actually need this default any more? Can we just add this as the configured value in all environments

Comment (sam803) [reply libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java]: Yeh I guess I was being a bit cautious but have added them here https://github.com/govuk-one-login/ipv-core-common-infra/pull/1026 , will remove

**Review by Joe-Edwards-GDS — DISMISSED** [review]
Thanks - didn't realise it'd be so noisy but looks good :)

Obviously don't merge before the config itself has been deployed!

**Review by Joe-Edwards-GDS — APPROVED** [review]
Nice - thanks for updating the other ones as well!

**Review by Joe-Edwards-GDS — CHANGES_REQUESTED** [review]
Ok looks like you need to add the config to the local running params as well, but hopefully obvious how to do that...

File: local-running/core.local.params.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Local running config params
+œ# Local running config params
```
Comment (Joe-Edwards-GDS) [inline local-running/core.local.params.yaml]: 👀

Comment (sam803) [reply local-running/core.local.params.yaml]: thanks, not sure where that came from :)

**Review by Joe-Edwards-GDS — APPROVED** [review]

**Pull request merged**